src/datagrid/DataGridFilter.tsx

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as React from "react";
+import {classNames} from "../utils";
+import {ClassNames} from "./ClassNames";
+import {Button} from "../forms/button";
+import {DataGridRow} from "./DataGrid";
+
+type DataGridFilterProps = {
+    style?: any;
+    datagridRef: any;
+    columnName: string;
+    onFilter: (rows: DataGridRow[], columnValue: any, columnName: string) => DataGridRow[];
+};
+
+type DataGridFilterState = {
+    isOpen: boolean;
+    filterValue: any;
+};
+
+export class DataGridFilter extends React.PureComponent<DataGridFilterProps, DataGridFilterState> {
+    private refParent = React.createRef<HTMLDivElement>();
+    private refChild = React.createRef<HTMLDivElement>();
+
+    state: DataGridFilterState = {
+        isOpen: false,
+        filterValue: undefined,
+    };
+
+    handleButtonClick = () => {
+        this.toggle();
+    };
+
+    toggle(isOpen = !this.state.isOpen) {
+        this.setState({isOpen}, this.afterToggle);
+    }
+
+    afterToggle = () => {
+        if (this.state.isOpen) {
+            this.subscribeDocumentClick();
+        } else {
+            this.unsubscribeDocumentClick();
+        }
+    };
+
+    subscribeDocumentClick = () => {
+        window.addEventListener("click", this.handleDocumentClick as any, true);
+    };
+
+    unsubscribeDocumentClick = () => {
+        window.removeEventListener("click", this.handleDocumentClick as any, true);
+    };
+
+    handleDocumentClick = (evt: React.MouseEvent<HTMLElement>) => {
```

Comment:
handleDocumentClick can be refactored as below.
handleDocumentClick = (evt: Event) => {
            if (this.state.isOpen) {
                const target = (evt.target as any) as HTMLElement;
                const el = this.refChild.current;

                if (el && typeof el !== "string" && !el.contains(target)) {
                    this.toggle();
                }
            }
        };

Reply:
done